### PR TITLE
ydcv 0.4 (new formula)

### PR DESCRIPTION
**Pull request by ladrift**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
A very useful command-line English-to-Chinese dictionary.

### CODE REVIEW

File: Formula/ydcv.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Ydcv < Formula
+  desc "YouDao Console Version - Simple wrapper for Youdao online translate (Chinese <-> English) service API, as an alternative to the StarDict Console Version(sdcv)."
+  homepage "https://github.com/felixonmars/ydcv"
+  url "https://github.com/felixonmars/ydcv/archive/0.4.tar.gz"
+  sha256 "2d9f6309bbf2d35c0c34c5ee945cf40769cc8201e6f374fa2a4f2d4b827fbdbb"
+
+  depends_on "python3"
```
Comment (fxcoudert) [inline Formula/ydcv.rb]: Github page states that Python 2.7 is supported

File: Formula/ydcv.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class Ydcv < Formula
+  desc "YouDao Console Version - Simple wrapper for Youdao online translate (Chinese <-> English) service API, as an alternative to the StarDict Console Version(sdcv)."
```
Comment (fxcoudert) [inline Formula/ydcv.rb]: Description should be 70 characters maximum

File: Formula/ydcv.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+  depends_on "python3"
+
+  def install
+    bin.install "ydcv.py" => "ydcv"
+    zsh_completion.install "contrib/zsh_completion" => "_ydcv"
+  end
+
+  test do
+    system "#{bin}/ydcv", "hello"
```
Comment (fxcoudert) [inline Formula/ydcv.rb]: Please check the output

**Review by fxcoudert — DISMISSED** [review]

File: Formula/ydcv.rb (hunk 1 above)
Comment (ladrift) [reply Formula/ydcv.rb]: The macOS default Python interpreter's version is 2.7.10. Should I add a `depends_on "python"` here?

Comment (fxcoudert) [reply Formula/ydcv.rb]: `depends_on :python if MacOS.version <= :snow_leopard`

File: Formula/ydcv.rb (hunk 2 above)
Comment (ladrift) [reply Formula/ydcv.rb]: Done

File: Formula/ydcv.rb (hunk 1 above)
Comment (ladrift) [reply Formula/ydcv.rb]: Done

File: Formula/ydcv.rb (hunk 3 above)
Comment (ladrift) [reply Formula/ydcv.rb]: Done

**Comment by ladrift** [comment]
Ping @fxcoudert. I have already complete the requested changes. Can this formula be added?

File: Formula/ydcv.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  sha256 "2d9f6309bbf2d35c0c34c5ee945cf40769cc8201e6f374fa2a4f2d4b827fbdbb"
+
+  depends_on :python if MacOS.version <= :snow_leopard
+
+  def install
+    bin.install "ydcv.py" => "ydcv"
+    zsh_completion.install "contrib/zsh_completion" => "_ydcv"
+  end
+
+  test do
```
Comment (ilovezfs) [inline Formula/ydcv.rb]:
```
    assert_match "hello", shell_output("#{bin}/ydcv 你好")
    assert_match "你好", shell_output("#{bin}/ydcv hello")
```

**Review by ilovezfs — CHANGES_REQUESTED** [review]
Please mark this formula `bottle :unneeded`

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @ydcv! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request merged**